File: 3936-split-array-by-prime-indices/3936-split-array-by-prime-indices.cpp
class Solution {
public:

    bool isPrime(int num){
        if(num == 1 || num == 0) return false;

        for(int i=2;i<=sqrt(num);i++){
            if(num % i == 0) return false;
        }
        return true;
    }

    long long splitArray(vector<int>& nums) {

        long long sumA = 0, sumB = 0;

        for(int i=0;i<nums.size();i++){
            if(isPrime(i)){
                sumA += nums[i];
            }
            else{
                sumB += nums[i];
            }
        }

        return abs(sumA - sumB);
    }
};

File: 3936-split-array-by-prime-indices/3936-split-array-by-prime-indices.java
class Solution {

    boolean isPrime(int num){
        if(num == 0 || num == 1) return false;

        for(int i=2;i<=Math.sqrt(num);i++){
            if(num % i == 0) return false;
        }
        return true;
    }

    public long splitArray(int[] nums) {
        
        long sumA = 0,sumB = 0;

        for(int i=0;i<nums.length;i++){
            if(isPrime(i)){
                sumA += nums[i];
            }
            else{
                sumB += nums[i];
            }
        }

        return Math.abs(sumA - sumB);
    }
}